File: Codewars/4kyu/roman-numerals-encoder/Python/solutions2.py
# Python - 2.7.6

# 列出所有可能的進位值
symbols = [
    (1000, 'M'), 
    (900, 'CM'), (500, 'D'), (400, 'CD'), (100, 'C'), 
    (90, 'XC'), (50, 'L'), (40, 'XL'), (10, 'X'), 
    (9, 'IX'), (5, 'V'), (4, 'IV'), (1, 'I')
]

def solution(n):
    result = ''
    for number, roman in symbols:
        result += roman * (n // number) # 符號出現次數
        n %= number                     # 剩餘位數
    return result

File: Codewars/4kyu/roman-numerals-encoder/Ruby/solutions2.rb
# Ruby - MRI 2.3.0

def solution(n)
    # 列出所有可能的進位值
    symbols = {
        M: 1000, 
        CM: 900, D: 500, CD: 400, C: 100, 
        XC: 90, L: 50, XL: 40, X: 10, 
        IX: 9, V: 5, IV: 4, I: 1
    }
    
    result = ""
    symbols.each do |roman, number|
        result += roman.to_s * (n / number) # 符號出現次數
        n %= number                         # 剩餘位數
    end
    
    return result
end